Remove another batch of deprecations

Hlavtox opened the pull request:
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Removes another big batch of deprecations, details below.
| Type?             | refacto
| Category?         | CO
| BC breaks?        | yes
| Deprecations?     | no
| How to test?      | Tests green, it should reveal any issue with making an order, registering, stock registration etc.
| UI Tests          | https://github.com/Hlavtox/ga.tests.ui.pr/actions/runs/7812271000
| Fixed issue or discussion?     | 
| Related PRs       | 
| Sponsor company   | 

### Description
- I completely removed the remains of advanced stock management and the old stock system. Along with all remains of supply orders, warehouses and unused ps_stock table.
- Removed some unused and deprecated properties on products.

### Removed methods and changes
classes/Address.php - deleted id_warehouse property
classes/Configuration.php - deleted clearConfigurationCacheForTesting method
classes/ObjectModel.php - deleted unused condition of passwd field
classes/Product.php - removed properties tax_name, base_price, id_color_default, advanced_stock_management, depends_on_stock
classes/order/OrderDetail.php - removed tax_name and id_warehouse properties
classes/pdf/PDF.php - removed unused TEMPLATE_SUPPLY_ORDER_FORM constant
classes/stock/StockAvailable.php - removed depends_on_stock property
classes/stock/StockMvt.php - cleaned up, fixed stock reference
classes/stock/StockMvtReason.php - cleaned up
classes/webservice/WebserviceRequest.php - fixed class reference
controllers/admin/AdminImportController.php - removed supplyOrdersImport and supplyOrdersDetailsImport methods
controllers/admin/AdminPdfController.php - removed processGenerateSupplyOrderFormPDF method
src/Core/Grid/Query/ManufacturerAddressQueryBuilder.php - id_warehouse is now removed
src/PrestaShopBundle/Entity/StockMvt.php - removed unused columns 

### Removed classes and files
classes/WarehouseAddress.php
classes/lang/SupplyOrderStateLang.php
classes/pdf/HTMLTemplateSupplyOrderForm.php
classes/stock/Stock.php
classes/stock/StockManager.php
classes/stock/StockManagerFactory.php
classes/stock/StockManagerInterface.php
classes/stock/StockManagerModule.php
classes/stock/StockMvtWS.php
classes/stock/SupplyOrder.php
classes/stock/SupplyOrderDetail.php
classes/stock/SupplyOrderHistory.php
classes/stock/SupplyOrderReceiptHistory.php
classes/stock/SupplyOrderState.php
classes/stock/Warehouse.php
classes/stock/WarehouseProductLocation.php
docs/csv_import/supply_orders_details_import.csv
docs/csv_import/supply_orders_import.csv
install-dev/data/xml/warehouse.xml
pdf/supply-order.addresses-tab.tpl
pdf/supply-order.product-tab.tpl
pdf/supply-order.tax-tab.tpl
pdf/supply-order.total-tab.tpl
pdf/supply-order.tpl
pdf/supply-order-footer.tpl
pdf/supply-order-header.tpl

### Database changes required for upgraded installs
- address
  - removed id_warehouse column
  - removed id_warehouse primary key
- order_detail
  - removed id_warehouse column
  - removed tax_name column
- product
  - removed advanced_stock_management column
- product_shop
  - removed advanced_stock_management column
- stock
  - completely removed
- warehouse
  - completely removed
- warehouse_product_location
  - completely removed
- warehouse_shop
  - completely removed
- warehouse_carrier
  - completely removed
- stock_available
  - removed depends_on_stock column
- stock_mvt
  - removed id_supply_order column
  - removed price_te column
  - removed last_wa column
  - removed current_wa column
  - removed referer column  
- supply_order
  - completely removed
- supply_order_detail
  - completely removed
- supply_order_history
  - completely removed
- supply_order_state
  - completely removed
- supply_order_state_lang
  - completely removed
- supply_order_receipt_history
  - completely removed

Hlavtox committed:
Remove big batch of deprecations

FabienPapet reviewed (dismissed):
Less code, less bugs, nice work 😊

matks reviewed (changes requested):
Disagree with this PR.

1. Some of the things you removed are marked `@deprecated since 9.0 and will be removed in 10.0, stock is now managed by new logic` and there is reason why. Why do you just ignore that.
2. You remove things without checking further whether or not they are being used. See `ajaxDie()` https://github.com/search?q=org%3APrestaShop%20ajaxDie&type=code

I might be wrong but it feels like this PR is just a throw-away PR where you thought "today I feel like I could clean some code" and you go here and there, removing things, without worrying about the impact, confident that nothing could go wrong :/

No jeopardizing the project because "it feels cleaner" please :/

matks commented:
@FabienPapet @FabienPapet Did you really review the PR :/ ? Because you approve a PR where multiple files marked `@deprecated since 9.0 and will be removed in 10.0` are deleted in the `develop` branch. Yes I guess you can miss one or two, but there are more than ten files like this

Hlavtox commented:
@matks I removed them because I was the one who put the deprecated notice. The ASM stuff was deprecated and non functional for the whole lifetime of 1.7, so I said that it's better to remove it fully instead of having non working object model and tables there. 🤷‍♂️

Regarding the ajaxDie and all other methods - I have checked for usage of EVERYTHING I removed and it's not used by the core. The usages in native modules will be fixed in time. The SA developer team does the same things - remove first and then fix the usages.

Hlavtox commented:
Update - I have extracted the ajax related things to separate PR.

Hlavtox committed:
Fix CS

jolelievre commented on classes/ObjectModel.php:
```diff
@@ -86,14 +86,14 @@ abstract class ObjectModelCore implements PrestaShop\PrestaShop\Core\Foundation\
 
     /**
-     * @deprecated 1.5.0.1 Define property using $definition['table'] property instead.
+     * @deprecated since 1.5.0.1 Define property using $definition['table'] property instead.
```
Why not remove it then?

Hlavtox replied:
I just wanted to make searching for `deprecated since` easier. :-)

jolelievre commented on classes/Product.php:
```diff
@@ -508,7 +470,7 @@ class ProductCore extends ObjectModel
             /*
              * Only the DB field is deprecated because unit_price is the new reference, we need to keep the class field though
-             * @deprecated in 8.0 this DB column will be removed in a future version
+             * @deprecated since 8.0 this DB column will be removed in a future version
```
Same here maybe it's time to remove it? Maybe in a dedicated PR though because there are other places that need to be clean (where we update this value in some arrays for example)

Hlavtox replied:
I just wanted to make searching for `deprecated since` easier. :-)

jolelievre commented on classes/controller/FrontController.php:
```diff
@@ -1109,7 +1109,7 @@ public function unregisterJavascript($id)
 
     /**
-     * @deprecated 1.7 This function shouldn't be used, use $this->registerStylesheet() instead
+     * @deprecated since 1.7 This function shouldn't be used, use $this->registerStylesheet() instead
```
Oh I didn't know this one had been deprecated 🤔
It's surprising because it's still used a lot, we even have an issue to fix this in BO because of modification related to the Symfony layout

Hlavtox replied:
Yep, for 1.7, you need to use a different method.
However, in the controller parent and BO, it's not marked as deprecated and actively used.

jolelievre commented on classes/pdf/HTMLTemplateSupplyOrderForm.php:
```diff
@@ -1,261 +0,0 @@
-
-/**
- * @since 1.5
- * @deprecated since 9.0 and will be removed in 10.0, stock is now managed by new logic
```
It's been deprecated in 9.0 only, so it shouldn't be removed now but in 10.0 as written in the comment

Hlavtox replied:
Yes

jolelievre commented on classes/pdf/PDF.php:
```diff
@@ -64,9 +64,6 @@ class PDFCore
     public const TEMPLATE_ORDER_SLIP = 'OrderSlip';
     public const TEMPLATE_DELIVERY_SLIP = 'DeliverySlip';
 
-    /** @deprecated since 9.0 and will be removed in 10.0 **/
```
Too soon

Hlavtox replied:
Yes

jolelievre commented on classes/stock/Stock.php:
```diff
@@ -1,194 +0,0 @@
- * Represents the products kept in warehouses.
- *
- * @since 1.5.0
- * @deprecated since 9.0 and will be removed in 10.0, stock is now managed by new logic
```
Too soon

Hlavtox replied:
Agree

jolelievre commented on classes/stock/StockManager.php:
```diff
@@ -1,842 +0,0 @@
- * StockManager : implementation of StockManagerInterface.
- *
- * @since 1.5.0
- * @deprecated since 9.0 and will be removed in 10.0, stock is now managed by new logic
```
Too soon

Hlavtox replied:
Agree

jolelievre commented on classes/stock/StockManagerFactory.php:
```diff
@@ -1,82 +0,0 @@
- * StockManagerFactory : factory of stock manager
- *
- * @since 1.5.0
- * @deprecated since 9.0 and will be removed in 10.0, stock is now managed by new logic
```
Too soon

Hlavtox replied:
Agree

jolelievre commented on classes/stock/StockManagerInterface.php:
```diff
@@ -1,131 +0,0 @@
- * StockManagerInterface : defines a way to manage stock.
- *
- * @since 1.5.0
- * @deprecated since 9.0 and will be removed in 10.0, stock is now managed by new logic
```
Too soon

Hlavtox replied:
Agree

jolelievre commented on classes/stock/StockManagerModule.php:
```diff
@@ -1,61 +0,0 @@
-
-/**
- * @since 1.5.0
- * @deprecated since 9.0 and will be removed in 10.0, stock is now managed by new logic
```
Too soon

Hlavtox replied:
Agree

jolelievre commented on classes/stock/StockMvt.php:
```diff
@@ -25,10 +25,11 @@
+ *
+ * @since 1.5.0
  */
-class StockMvtCore extends ObjectModel
```
Hmm no I think it makes sense to extend ObjectModel, this way the override feature allows to override the base class of all object models Although it's not recommended it still makes sense for this override feature

jolelievre commented on classes/stock/StockMvt.php:
```diff
@@ -81,47 +74,10 @@ class StockMvtCore extends ObjectModel
-     *
-     * @var int Used when the movement is due to a supplier order
-     */
-    public $id_supply_order = null;
```
Why remove those particular fields? They were not marked as deprecated

Hlavtox replied:
Because id_supply_order is related to supply orders / advanced stock management, that was removed long time ago. This is just cleaning more remains.

jolelievre commented on classes/stock/StockMvt.php:
```diff
@@ -136,110 +92,24 @@ class StockMvtCore extends ObjectModel
-     *
-     * @return array mvts
-     */
-    public static function getNegativeStockMvts($id_order, $id_product, $id_product_attribute, $quantity, $id_warehouse = null)
```
SAme here why remove this code?

Hlavtox replied:
Whole class is to be removed, unused.

jolelievre commented on classes/stock/StockMvtReason.php:
```diff
@@ -25,7 +25,7 @@
  */
 
 /**
- * @deprecated since 9.0 and will be removed in 10.0, this object model is no longer needed
```
So it's not deprecated anymore?

Hlavtox replied:
Yes, I marked it as deprecated by accident I think.

jolelievre commented on classes/stock/StockMvtWS.php:
```diff
@@ -1,305 +0,0 @@
- * Webservice entity for stock movements.
- *
- * @since 1.5.0
- * @deprecated since 9.0 and will be removed in 10.0, this object model is no longer needed
```
Too soon

Hlavtox replied:
Agree

jolelievre commented on classes/stock/SupplyOrder.php:
```diff
@@ -1,584 +0,0 @@
-
-/**
- * @since 1.5.0
- * @deprecated since 9.0 and will be removed in 10.0
```
Too soon

Hlavtox replied:
Agree

jolelievre commented on classes/stock/SupplyOrderDetail.php:
```diff
@@ -1,295 +0,0 @@
- * Represents one product ordered.
- *
- * @since 1.5.0
- * @deprecated since 9.0 and will be removed in 10.0
```
Too soon

Hlavtox replied:
Agree

jolelievre reviewed (changes requested):
I didn't review everything but you are removing some classes/code that are not deprecated

Can you please revert those changes? I'll review again when the PR is a bit cleaner
